### PR TITLE
Add `accounts` global to truffle console

### PR DESCRIPTION
**Pull request by pensivej**
This PR relates to #2157 
Also, thanks to @gnidan for quick comments :)
I set the branch somehow complicated so I just removed the fork and made the new one 👍 so that I can submit two separate PRs.

### CODE REVIEW

**Comment by eggplantzzz** [comment]
This looks nice. Nice and simple.

File: packages/truffle-core/lib/console.js [hunk 1]
```diff
@@ -52,13 +52,16 @@ class Console extends EventEmitter {
...
     try {
       const abstractions = this.provision();
-      this.repl.start({
-        prompt: "truffle(" + this.options.network + ")> ",
-        context: {
-          web3: this.web3
-        },
-        interpreter: this.interpret.bind(this),
-        done: callback
+      this.web3.eth.getAccounts().then(accounts => {
```
Comment (gnidan) [inline packages/truffle-core/lib/console.js]: Oh shoot. I might be being paranoid, but I think maybe we want this entire function to get wrapped in the `.then()`. I'm not sure, but we might need to make sure that `this.resetContractsInConsoleContext` doesn't rely on some implicit ordering. _shrug_ I hate these callback-style functions everywhere, it makes the code so ugly :(

Comment (pensivej) [reply packages/truffle-core/lib/console.js]: Maybe including this.resetContractsInConsoleContext into .then() will solve the issue?
I don't like this type of callback functions everywhere also :(

**Comment by fainashalts** [comment]
+1 This looks good to me!

**Comment by eggplantzzz** [comment]
Cool, time to merge!

**Comment by pensivej** [comment]
Thanks for the review :)

**Pull request merged**